File: src/main/java/org/telran/lecture_14_hash/hw/Task02.js
// Дан массив целых чисел.
// Найти первое вхождение элемента в массиве, который встречается ровно k раз.

// Функция находит первый элемент в массиве, который имеет частоту встречаемости, равную заданному числу k.
// Если такого элемента в массиве нет, функция возвращает -1.
function contains(arr, el){
    for (let i = 0; i < arr.length; i++){
        if (el == arr[i]){return true}
    }
    return false;
}

function firstElement(numbers, k) {
    let frequency = {};
    for (let i = 0; i < numbers.length; i++) {
        if (contains(Object.keys(frequency), numbers[i])){
            frequency[numbers[i]] = frequency[numbers[i]] + 1;
        }else{
            frequency[numbers[i]] = 1;
        }
    }
    for (let i = 0; i < numbers.length; i++){
        if (frequency[numbers[i]] === k){
            return numbers[i];
        }
    }
    // console.log(frequency);
    return -1;
}

// Вариант-2, через Map()
function firstElement(numbers, k) {
    const frequencyMap = new Map();

    // Подсчитываем частоту каждого элемента
    for (let num of numbers) {
        frequencyMap.set(num, (frequencyMap.get(num) || 0) + 1);
    }

    // Ищем первый элемент, который встречается ровно k раз
    for (let num of numbers) {
        if (frequencyMap.get(num) === k) {
            return num;
        }
    }

    return -1;  // Если такого элемента нет
}

let numbers = [2, 8, 2, 0, 4, 6, 4, 10, 8];
let k = 3;


result = firstElement(numbers, k);
console.log(result);

File: src/main/java/org/telran/lecture_14_hash/hw/Task02.java
package org.telran.lecture_14_hash.hw;
import java.util.HashMap;
import java.util.Map;
// Дан массив целых чисел.
// Найти первое вхождение элемента в массиве, который встречается ровно k раз.

// Функция находит первый элемент в массиве, который имеет частоту встречаемости, равную заданному числу k.
// Если такого элемента в массиве нет, функция возвращает -1.

public class Task02 {
    public static void main(String[] args) {
        Integer[] numbers = { 8, 2, 0, 4, 6, 4, 10, 8};
        int k = 4;
        int result = firstElement(numbers, k);
        System.out.println(result);
    }

    public static int firstElement(Integer[] numbers, int k) {
        Map<Integer, Integer>frequency = new HashMap<>();
        for (Integer number : numbers) {
            frequency.put(number, frequency.getOrDefault(number, 0) + 1);
        }
        for (Integer number : numbers){
            if (frequency.get(number) == k) {
                return number;
            }
        }
        return -1;
    }
}
